app_dart/lib/src/request_handlers/github/webhook_subscription.dart

```diff
@@ -0,0 +1,659 @@
+        // and schedule new ones.
+        await _scheduleIfMergeable(pullRequestEvent);
+        break;
+      // Ignore the rest of the events.
+      case 'ready_for_review':
+      case 'unlabeled':
+      case 'assigned':
+      case 'locked':
+      case 'review_request_removed':
+      case 'review_requested':
+      case 'unassigned':
+      case 'unlocked':
+        break;
+    }
+  }
+
+  /// This method assumes that jobs should be cancelled if they are already
+  /// runnning.
+  Future<void> _scheduleIfMergeable(
+    PullRequestEvent pullRequestEvent,
+  ) async {
+    final PullRequest pr = pullRequestEvent.pullRequest!;
+    final RepositorySlug slug = pullRequestEvent.repository!.slug();
+
+    log.info(
+        'Scheduling tasks if mergeable(${pr.mergeable}): owner=${slug.owner} repo=${slug.name} and pr=${pr.number}');
+
+    // The mergeable flag may be null. False indicates there's a merge conflict,
+    // null indicates unknown. Err on the side of allowing the job to run.
+    if (pr.mergeable == false) {
+      final RepositorySlug slug = pullRequestEvent.repository!.slug();
+      final GitHub gitHubClient = await config.createGitHubClient(pullRequest: pr);
+      final String body = config.mergeConflictPullRequestMessage;
+      if (!await _alreadyCommented(gitHubClient, pr, body)) {
+        await gitHubClient.issues.createComment(slug, pr.number!, body);
+      }
+
+      return;
+    }
+
+    await scheduler.triggerPresubmitTargets(pullRequest: pr);
+  }
+
+  /// Release tooling generates cherrypick pull requests that should be granted an approval.
+  Future<void> _tryReleaseApproval(
+    PullRequestEvent pullRequestEvent,
+  ) async {
+    final PullRequest pr = pullRequestEvent.pullRequest!;
+    final RepositorySlug slug = pullRequestEvent.repository!.slug();
+
+    final String defaultBranch = Config.defaultBranch(slug);
+    final String? branch = pr.base?.ref;
+    if (branch == null || branch.contains(defaultBranch)) {
+      // This isn't a release branch PR
+      return;
+    }
+
+    final List<String> releaseAccounts = await config.releaseAccounts;
+    if (releaseAccounts.contains(pr.user?.login) == false) {
+      // The author isn't in the list of release accounts, do nothing
+      return;
+    }
+
+    final GitHub gitHubClient = config.createGitHubClientWithToken(await config.githubOAuthToken);
+    final CreatePullRequestReview review = CreatePullRequestReview(slug.owner, slug.name, pr.number!, 'APPROVE');
+    await gitHubClient.pullRequests.createReview(slug, review);
+  }
+
+  Future<void> _checkForLabelsAndTests(PullRequestEvent pullRequestEvent) async {
+    final PullRequest pr = pullRequestEvent.pullRequest!;
+    final String? eventAction = pullRequestEvent.action;
+    final String repo = pr.base!.repo!.fullName.toLowerCase();
+    if (kNeedsCheckLabelsAndTests.contains(repo)) {
+      final GitHub gitHubClient = await config.createGitHubClient(pullRequest: pr);
+      try {
```

Comment:
I am assuming that we do not catch the exception because it trickles back to the call of the REST API. Is that correct?

Reply:
Correct!